demos/mqtt/mqtt_demo_basic_tls/demo_config.h

```diff
@@ -50,27 +50,22 @@
 /**
  * @brief MQTT server host name.
  *
- * This demo uses the Mosquitto test server. This is a public MQTT server; do not
- * publish anything sensitive to this server.
+ * This demo can be run using the open-source Mosquitto server.
  * Mosquitto MQTT broker can run locally as an alternate option. Please refer to
  * the instructions in https://mosquitto.org/ for running a Mosquitto broker
  * locally.
+ * Instructions to run Mosquitto on Docker container can be viewed in the
+ * README.md of the root directory.
+ *
+ * #define BROKER_ENDPOINT               "...insert here..."
```

Comment:
I see value in being able to run the demos quickly without having the change a bunch of demo_config.h files. If we recommend running a broker locally, why not just change all of these to "localhost" instead?

Reply:
Not sure whether that is necessarily helpful in speeding up the experience of running demos. With `localhost`, the customer would need to have a broker setup on the _same_ machine. (The customer may very well want to setup their broker on a remote host). 
A default value would be more valuable in speeding up the customer experience of running the demo, if the broker is publicly available (as was the case with `test.mosquitto.org`). 
I can instead update the comments to use `localhost` instead of "...insert here..."

Reply:
If they want to set up a broker on a remote host, that's their choice, just like they were able to choose to not use `test.mosquitto.org` before this change. Our wording of running the broker "locally" implies that they'd run it on the same machine as the posix demos, so `localhost` would be an obvious choice for a default. 

If we want these demos to not build without editing these files or without specifying a cmake flag, then we don't need to make any changes; I'm just pointing out that making the change would mean one less cmake flag/one less file to edit for someone who is following our instructions on setting up a local broker

Reply:
Have added the default configuration back, as I don't strongly feel against it.

Reply:
Is the mosquitto broker installed and running by default on a machine? If not, when I run this demo , I will get connection refused error - at which point I need to go and debug why the connection was refused and eventually realize the that I do not have a mosquitto broker setup. 

Wouldn't a #error be better telling me that I need to setup mosquitto broker.

Reply:
Yes, I agree that having the default value of `localhost` can cause more developer effort in identifying the failure cause. Will remove the default value in a separate PR